packages/vitest-runner/src/index.ts

```diff
@@ -0,0 +1,49 @@
+import { getV8Flags, Measurement } from "@codspeed/core";
+import { ConfigEnv, defineConfig, UserConfig } from "vitest/config";
+
+type VitestConfig = ReturnType<typeof defineConfig>;
+
+function applyCodSpeedConfig(config: UserConfig): UserConfig {
+  // TODO: prevent opting out of `pool: 'forks'` and/or using `poolOptions.forks.singleFork`
+  return {
+    ...config,
+    test: {
+      ...config.test,
+      pool: "forks",
+      poolOptions: {
+        forks: {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore - remove when https://github.com/vitest-dev/vitest/pull/4383 is merged and released
+          execArgv: getV8Flags(),
+        },
+      },
+      runner: `${__dirname}/runner.es5.js`,
+      globalSetup: [
+        ...(config.test?.globalSetup ?? []),
+        `${__dirname}/globalSetup.es5.js`,
+      ],
+    },
+  };
+}
+
+export async function withCodSpeed(
```

Comment:
I would recommend implementing this as a plugin since this is how config is usually modified in the ecosystem:

```ts
// vitest.config.ts
import { CodeSpeedPlugin } from '@codespeed/vitest-runner'
import { defineConfig } from 'vitest/config'

export default defineConfig({
  plugins: [CodeSpeedPlugin()]
})
```

```ts
// plugin.ts
export default () => {
  return {
    name: 'codespeed:vitest'
    config() {
      // you don't need to merge it yourself, just return modified version
      return {
        test: {
          poolOptions: {
          forks: {
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore - remove when https://github.com/vitest-dev/vitest/pull/4383 is merged and released
              execArgv: getV8Flags(),
            },
          },
          runner: `${__dirname}/runner.es5.js`,
          globalSetup: [
            `${__dirname}/globalSetup.es5.js`,
          ],
        }
      }
    }
  }
}
```

Reply:
Thanks for the suggestion, I implemented it in the new PR #28